System/Win32/Registry.hsc

```diff
@@ -402,19 +399,21 @@ regQueryValueKey key mb_subkey =
 foreign import WINDOWS_CCONV unsafe "windows.h RegQueryValueW"
   c_RegQueryValue :: PKEY -> LPCTSTR -> LPTSTR -> Ptr LONG -> IO ErrCode
 
-regQueryValue :: HKEY -> String -> IO String
-regQueryValue key mb_subkey =
```

Comment:
So the plan is to deprecate it in the next release (telling users to use the new `regQueryDefaultValue`) , but then un-deprecate it after that, having changed the implementation?

Reply:
Yes and then I can add it to the migration guide for GHC 8.6, though if you plan to make the replacement function already in this version we can drop the deprecated pragma later with the new implementation PR, as long as there's an interface change. Luckily it looks like the second parameter should be a `Maybe String`.

Reply:
OK, I had changed the signature and implementation of `regQueryValue` all in one here.  I don't think I can write the replacement function now because the replacement should also be named `regQueryValue`, to match the API function.  So I'll just deprecate it with a note now.

Reply:
I'm slightly confused, the documentation for the `regQueryValue` function suggests that a replacement function with the same name would have a different signature https://msdn.microsoft.com/en-us/library/windows/desktop/ms724909(v=vs.85).aspx

e.g. the second parameter should be a `Maybe`. So you should be able to.

Reply:
Me too! 😄 The result of this PR does still contain a function named `regQueryValue`, with a new implementation and the signature changed from the old version such that the second parameter is `Maybe String`, so anybody using `regQueryValue` at the moment would get a compile error when they update (and there is no deprecation warning).  Perhaps it's the diff that's confusing, or probably I'm missing something!  I have an incoming update addressing most of your comments

Reply:
Oh sorry, I completely missed that! the functions are a bit close in name :) Ok in that case just drop the deprecate pragmas and I'll accept the changes :)

Reply:
At the risk of adding more confusion, this PR adds one deprecation pragma on `regQueryValueKey`, as the new implementation of `regQueryValue` is identical to it.  Do you want me to remove the deprecation pragma on `regQueryValueKey`, remove `regQueryValueKey` entirely, or something else?

Reply:
I'm sorry, It's all fine, I really shouldn't review code at 4am...